xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -444,6 +555,177 @@ public void run() {
         });
       }
     }
+
+    @Nullable
+    private Long determineFaultDelayNanos(FaultDelay faultDelay, Metadata headers) {
+      Long delayNanos;
+      Integer delayRate = null;
+      if (faultDelay.headerDelay()) {
+        try {
+          int delayMillis = Integer.parseInt(headers.get(HEADER_DELAY_KEY));
+          delayNanos = TimeUnit.MILLISECONDS.toNanos(delayMillis);
+          String delayPercentageStr = headers.get(HEADER_DELAY_PERCENTAGE_KEY);
+          if (delayPercentageStr != null) {
+            int delayPercentage = Integer.parseInt(delayPercentageStr);
+            delayRate = delayPercentage * 100 * 100;
```

Comment:
The default percentage’s denominator is HUNDRED if you go to the link on ` percentage` in the doc.

```
HUNDRED
(DEFAULT) ⁣100.
```

I'd be surprised if the percentage is computed with header's value as numerator and FaultDelay percentage's denominator as denominator. When the client sends the header, it can not predict what the config's denominator will be set by the control plane. It does not make any sense.

Reply:
Oh, right. Looks I misinterpreted the doc. It should just the default 100.